src/completion.ml

```diff
@@ -0,0 +1,147 @@
+(** Completion procedure *)
+
+open Terms
+open Basics
+open Extra
+
+(** total strict order *)
+type 'a order = 'a -> 'a -> bool
```

Comment:
It is perhaps more convenient to use comparison functions:
```
type 'a cmp = 'a -> 'a -> int
(* [cmp x y] returns
- [0] when x=y
- [r<0] when x<y
- (r>0] when x>y *)
```
Then, there is no need to introduce gt, ge, etc.
```